### PR TITLE
Extend some tests to format_version=3

### PR DESCRIPTION
**Pull request by maysamyabandeh**
format_version=3 changes the format of SST index. This is however not being tested currently since tests only work with the default format_version which is currently 2. The patch extends the most related tests to also test for format_version=3.

### CODE REVIEW

File: db/db_test_util.cc [hunk 1]
```diff
@@ -428,6 +428,15 @@ Options DBTestBase::GetOptions(
       options.prefix_extractor.reset(NewNoopTransform());
       break;
     }
+    case kBlockBasedTableWithPartitionedIndexFormat3: {
+      table_options.index_type = BlockBasedTableOptions::kTwoLevelIndexSearch;
+      options.prefix_extractor.reset(NewNoopTransform());
+      // Format 3 changes the binary index format. Since partitioned index is a
+      // super-set of simple indexes, we are also using kTwoLevelIndexSearch to
+      // test this format.
+      table_options.format_version = 3;
```
Comment (siying) [inline db/db_test_util.cc]: We need to be restraint about total test run time. Platforms like Travis has very tight duration limit. Test scenario added here will run a lot of tests with this new setting. I'm not against adding it but just be sure that you've thought of the question that whether we really want to test all of them with this new setting.

Comment (maysamyabandeh) [reply db/db_test_util.cc]: Lets wait for travis run and verify the runtime.

File: tools/db_crashtest.py [hunk 2]
```diff
@@ -48,6 +48,7 @@
     "verify_checksum": 1,
     "write_buffer_size": 4 * 1024 * 1024,
     "writepercent": 35,
+    "format_version": lambda: random.randint(2, 3),
```
Comment (ajkr) [inline tools/db_crashtest.py]: the db_stress and db_crashtest parts lgtm

File: db/db_test_util.cc (hunk 1 above)
Comment (maysamyabandeh) [reply db/db_test_util.cc]: This is the comparison of runtime. The difference is not much and could actually be mostly noice:
Before:
```
 TEST_GROUP=platform_dependent
 19 min 46 sec
 TEST_GROUP=1
 30 min 31 sec
 TEST_GROUP=2
 22 min 57 sec
 TEST_GROUP=3
 18 min 9 sec
 TEST_GROUP=4
 14 min 38 sec
```
After:
```
 TEST_GROUP=platform_dependent
 16 min 21 sec
 TEST_GROUP=1
 33 min 34 sec
 TEST_GROUP=2
 19 min 54 sec
 TEST_GROUP=3
 23 min 18 sec
 TEST_GROUP=4
 15 min 33 sec
```

**Comment by maysamyabandeh** [comment]
The travis test failure seems irrelevant:
```
[ RUN      ] DBTest.GroupCommitTest
db/db_test.cc:2174: Failure
Expected: (TestGetTickerCount(options, WRITE_DONE_BY_OTHER)) > (0), actual: 0 vs 0
terminate called after throwing an instance of 'testing::internal::GoogleTestFailureException'
  what():  db/db_test.cc:2174: Failure
Expected: (TestGetTickerCount(options, WRITE_DONE_BY_OTHER)) > (0), actual: 0 vs 0
Received signal 6 (Aborted)
```

**Pull request closed without merging**